packages/mdc-slider/adapter.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+
+/**
+ * Adapter for MDC Slider.
+ *
+ * Defines the shape of the adapter expected by the foundation. Implement this
+ * adapter to integrate the Slider into your framework. See
+ * https://github.com/material-components/material-components-web/blob/master/docs/authoring-components.md
+ * for more information.
+ *
+ * @record
+ */
+class MDCSliderAdapter {
+  /**
+   * Returns true if className exists for the slider Element
+   * @param {string} className
+   * @return {boolean}
+   */
+  hasClass(className) {}
+
+  /**
+   * Adds a class to the slider Element
+   * @param {string} className
+   */
+  addClass(className) {}
+
+  /**
+   * Removes a class from the slider Element
+   * @param {string} className
+   */
+  removeClass(className) {}
+
+  /**
+   * Returns a string if attribute name exists on the slider Element,
+   * otherwise returns null
+   * @param {string} name
+   * @return {?string}
```

Comment:
Have Andy do a pass on your nulls and undefineds...He's more of an expert than me...and apparently these is a difference

Reply:
@acdvorak plz help me, i am uncertain :fearful: